uint8[] fooId
int8[3] fooPos
int8[] fooTimestamp
